janusgraph-core/src/main/java/org/janusgraph/graphdb/transaction/StandardJanusGraphTx.java

```diff
@@ -807,8 +811,15 @@ private void connectRelation(InternalRelation r) {
             if (!success) throw new AssertionError("Could not connect relation: " + r);
         }
         addedRelations.add(r);
-        for (int pos = 0; pos < r.getLen(); pos++) vertexCache.add(r.getVertex(pos), r.getVertex(pos).longId());
-        if (TypeUtil.hasSimpleInternalVertexKeyIndex(r)) newVertexIndexEntries.add((JanusGraphVertexProperty) r);
+        for (int pos = 0; pos < r.getLen(); pos++) {
+            InternalVertex vertex = r.getVertex(pos);
+            if (!vertex.isNew()) {
```

Comment:
Can you explain this change? Is it because if vertex is new, then it must already have been put into vertexCache?

Reply:
Yes new vertex placed into cache in two places makeSchemaVertex/addVertex, so this change avoid extra cache updates